pkg/agent/connect.go

```diff
@@ -18,6 +18,8 @@ type Conn struct {
 
 	Stream *yamux.Stream
 	Labels *pb.LabelSet
+
+	cleanup []func() error
```

Comment:
This is _fine_, only feedback is if you'd want to implement it similarly to: https://golang.org/src/testing/testing.go?s=30570:30604#L872

Basically: each one is a new closure using a `defer` so cleanup happens even if another cleanup panics.

Reply:
in this case, nah, because it's so isolated. But in the future maybe.

Reply:
This code was specifically a sort of double check closing too.